src/vm/tests/contracts.rs

```diff
@@ -486,6 +502,131 @@ fn test_contract_caller(owned_env: &mut OwnedEnvironment) {
     }
 }
 
+fn test_sponsor_type_is_optional(owned_env: &mut OwnedEnvironment) {
+    let p1 = execute("'SZ2J6ZY48GV1EZ5V2V5RB9MP66SW86PYKKQ9H6DPR");
+    let env = owned_env.get_exec_environment(Some(p1.clone()), Some(p1.clone()));
+}
+
+fn test_sponsor_type_is_optional_principal(owned_env: &mut OwnedEnvironment) {
+    let p1 = execute("'SZ2J6ZY48GV1EZ5V2V5RB9MP66SW86PYKKQ9H6DPR");
+    let sponsor = Value::some(Value::Bool(true)).unwrap();
+    let env = owned_env.get_exec_environment(Some(p1.clone()), Some(sponsor.clone()));
+}
+
+#[test]
+#[should_panic(
```

Comment:
I'm not sure I see why these are panicking failures?  Is it because the sender and sponsor are the same principal?

Reply:
Ah nm, I see -- it should be `(some SZ2J6ZY48GV1EZ5V2V5RB9MP66SW86PYKKQ9H6DPR)`.